### PR TITLE
Prep commits for having the MCD process encapsulated MachineConfig

### PR DESCRIPTION
**Pull request by cgwalters**
Prep commits for https://github.com/openshift/machine-config-operator/pull/904

### CODE REVIEW

File: pkg/daemon/daemon.go [hunk 1]
```diff
@@ -803,6 +791,53 @@ func (dn *Daemon) storeCurrentConfigOnDisk(current *mcfgv1.MachineConfig) error
...
+	if err == nil && ccAnnotation != "" {
+		return false, nil
+	}
+
+	glog.Infof("No %s annotation on node %s: %v, in cluster bootstrap, loading initial node annotation from %s", constants.CurrentMachineConfigAnnotationKey, dn.node.Name, dn.node.Annotations, constants.InitialNodeAnnotationsFilePath)
+
+	d, err := ioutil.ReadFile(constants.InitialNodeAnnotationsFilePath)
+	if err != nil && !os.IsNotExist(err) {
+		return false, fmt.Errorf("failed to read initial annotations from %q: %v", constants.InitialNodeAnnotationsFilePath, err)
+	}
```
Comment (runcom) [inline pkg/daemon/daemon.go]: super silly nit: shouldn't we return `true` from now on here? I'm just worried in the future we're gonna change the outer logic and might miss the bootstrap bool

File: pkg/daemon/update_test.go [hunk 2]
```diff
@@ -453,21 +451,3 @@ func checkIrreconcilableResults(t *testing.T, key string, reconcilableError erro
...
-	require.Nil(t, d.reboot("", 0, nil))
-
-	// skipReboot in normal cluster run is just a no-op and we reboot anyway
-	d = &Daemon{
-		mock:       true,
-		onceFrom:   "",
-		skipReboot: true,
-	}
-	require.NotNil(t, d.reboot("", 0, exec.Command("true")))
-}
```
Comment (runcom) [inline pkg/daemon/update_test.go]: I'm still reluctant to remove a test however easy and straightforward it is...if someone messes up with the skipReboot `if` this was going to catch it

Comment (cgwalters) [reply pkg/daemon/update_test.go]: We [talked about these unit tests before](https://github.com/openshift/machine-config-operator/pull/862#issuecomment-503264620) right?

Comment (runcom) [reply pkg/daemon/update_test.go]: yup, no blocking, was just pointing out my initial reaction when a test gets deleted :D

File: pkg/daemon/update.go [hunk 3]
```diff
@@ -169,14 +169,18 @@ func (dn *Daemon) update(oldConfig, newConfig *mcfgv1.MachineConfig) (retErr err
...
-	// reason is that there is a good chance a previous state is not available to match against.
-	if oldConfig.Kind == "" && dn.onceFrom != "" {
-		glog.Info("Missing kind in old config. Assuming no prior state.")
+	oldConfigUnset := oldConfig == nil
+	if oldConfigUnset {
+		// Rather than change the rest of the code to deal
+		// with a nil oldConfig, just pass an empty one in
+		// most places.
+		emptyMC := mcfgv1.MachineConfig{}
+		oldConfig = &emptyMC
```
Comment (ericavonb) [inline pkg/daemon/update.go]: note for next time (don't have to fix in this pr): can remove `emptyMC` declaration and just do `oldConfig = &mcfgv1.MachineConfig{}` directly

File: pkg/daemon/daemon.go (hunk 1 above)
Comment (ericavonb) [reply pkg/daemon/daemon.go]: Would that would apply for the `json.Unmarshal`  and `setNodeAnnotations(..., initial)` error cases as well?

File: pkg/daemon/daemon.go [hunk 4]
```diff
@@ -790,6 +793,53 @@ func (dn *Daemon) storeCurrentConfigOnDisk(current *mcfgv1.MachineConfig) error
...
+	if os.IsNotExist(err) {
+		// try currentConfig if, for whatever reason we lost annotations? this is super best effort.
+		currentOnDisk, err := dn.getCurrentConfigOnDisk()
+		if err == nil {
+			glog.Infof("Setting initial node config based on current configuration on disk: %s", currentOnDisk.GetName())
+			_, err := setNodeAnnotations(dn.kubeClient.CoreV1().Nodes(), dn.nodeLister, dn.node.Name, map[string]string{constants.CurrentMachineConfigAnnotationKey: currentOnDisk.GetName()})
```
Comment (ericavonb) [inline pkg/daemon/daemon.go]: why don't we update the cached copy `dn.node = n` like in the other case?

File: pkg/daemon/daemon.go (hunk 1 above)
Comment (runcom) [reply pkg/daemon/daemon.go]: yes, from this point it should be safe to return true

File: pkg/daemon/daemon.go [hunk 5]
```diff
@@ -1278,15 +1312,16 @@ func ValidPath(path string) bool {
 // senseAndLoadOnceFrom gets a hold of the content for supported onceFrom configurations,
 // parses to verify the type, and returns back the genericInterface, the type description,
 // if it was local or remote, and error.
-func (dn *Daemon) senseAndLoadOnceFrom() (interface{}, onceFromOrigin, error) {
+func (dn *Daemon) senseAndLoadOnceFrom(onceFrom string) (interface{}, onceFromOrigin, error) {
```
Comment (sinnykumari) [inline pkg/daemon/daemon.go]: Just wondering, since this function is not utilizing Daemon struct anymore, is there a reason to keep it part of Daemon ?

Comment (runcom) [reply pkg/daemon/daemon.go]: not really, we can remove the object reference to Daemon as a cleanup

**Comment by sinnykumari** [comment]
@cgwalters there are some conflict with latest master, can we get this PR rebased?

**Comment by cgwalters** [comment]
Eh, the reboot parts of this were merged, and there's been a lot of code churn since then that makes rebasing this tricky/painful.  I'm just going to try putting everything in https://github.com/openshift/machine-config-operator/pull/904 instead.

**Pull request closed without merging**